src/components/mc/cpu/mc_cpu.c

```diff
@@ -9,35 +9,142 @@
 #include "utils/ucc_malloc.h"
 #include <sys/types.h>
 
+static ucc_status_t ucc_mc_cpu_mem_alloc_with_init(ucc_mc_buffer_header_t **ptr,
+                                                   size_t size);
+
 static ucc_config_field_t ucc_mc_cpu_config_table[] = {
     {"", "", NULL, ucc_offsetof(ucc_mc_cpu_config_t, super),
      UCC_CONFIG_TYPE_TABLE(ucc_mc_config_table)},
 
-    {NULL}};
+    {"ELEM_SIZE", "1024", "The size of each element in mc cpu mpool",
```

Comment:
Since ELEM_SIZE and MAX_ELEMS are common parameters for both cpu/cuda components they can be moved to common config table: ucc_mc_base.c, ucc_mc_config_table

Reply:
@vspetrov I thought we purposely want them seperate in mc_cuda and mc_cpu, to be able to define different default values for each and to be able to configure them separately at runtime?

Reply:
they will be separate. this is how UCS_LOG works. There is a common config table: ucc_mc_base.c line 9: ucc_mc_config_table. There is a variable defined: LOG_LEVEL. This config table is then inherited (included) in mc/cpu and mc/cuda respectively. For example in mc/cpu: mc_cpu.c line 13. And by this inclusion the parameter will be defined in mc/cpu with its prefix, MC_CPU. so ucc_info will show 2 independent params for cpu and cuda: UCC_MC_CPU_LOG_LEVEL and UCC_MC_CUDA_LOG_LEVEL.

Reply:
yes, you are right, keep separate tables, then